graph.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
+# graph.py
+from collections import defaultdict
+
+
+class Graph():
+    def __init__(self, vertices):
+        self.graph = defaultdict(list)
+        self.V = vertices
+
+    def add_edge(self, u, v):
+        self.graph[u].append(v)
+
+    # internal
+    def _is_cyclic_util(self, v, visits_p, n):
+        visits = list(visits_p)
+        visits[v] += 1
+        bads = []
+        if(visits[v] > 1):
+            if(v == n):
```

Comment:
In general, preferred formatting is `if v == n:` (also for other `if`, `while`, etc. statements). Also, please give more descriptive names to your variables.